README.md

```diff
@@ -1,20 +1,106 @@
+|    R.2    | Rosetta API support with sub-account balances |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |                    |              |
+|    C.1    |   Cryptographic Verification of local state   |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |          X         |              |
+|    C.2    |  Cryptographic Proofs for remote state access |         X        |         X         |         X        |         X         |    X    |      X     |      X      |        X       |        X       |        X       |         X         |          X         |       X      |
+
+## Architecture
+
+### Components
+
+socket for live ones.)
+The Flow Data Provisioning Service is constituted of three main components.
+1. The **Chain** interface is responsible for reconstructing a view of the sequence of blocks, along with their metadata. It allows the consumer to step from the root block to the last sealed block, while presenting height, block identifier and state commitment for each step. The file i/o version does so by using the execution node's on-disk key-value store, while the network version relies on data retrieved from access nodes.
+2. The **Streamer** interface is responsible for streaming in-order trie updates from different sources; the file i/o version reads them from the LedgerWAL, while the network version receives trie updates through its network subscription on the execution node.
+3. The **Mapper** interface is responsible for mapping incoming state trie updates to blocks. Generally, trie updates come in by chunk, so each block maps from zero to multiple trie updates. Once a block is mapped to its respective trie updates, the mapper forwards the information to the indexer.
+4. The **Indexer** interface is responsible for receiving a set of trie updates for each block and creating the necessary main indexes and auxiliary in the on-disk database. These indexes allow efficient retrieval of the state at arbitrary block heights in the state history.
+5. The **Ledger** interface is responsible for providing clean APIs to access the execution state at different block heights and registers. Its implementations use the underlying indexes to emulate the same APIs as the Flow execution node Ledger, or the Rosetta Data API.
 
-- index transaction events
-- events retrieval interface
-- REST API interface
-- DPS web client
+### Diagram
 
-## Interfaces
+The following diagram describes a simple overview of an example where the DPS is reading from two sporks. One of them
+is a live one, which sends
 
-1. Indexer: indexes changes/deltas per block
-2. Update Stream: gets all the trie updates per chunk
-3. Chain View: has block sequence data
\ No newline at end of file
+```text
```

Comment:
Would be great, if @m4ksio could confirm that this works: 
<img width="595" alt="Screen Shot 2021-05-04 at 1 18 59 PM" src="https://user-images.githubusercontent.com/26322303/117064785-7f63e300-acdb-11eb-9a6a-4c118fb08e6c.png">

The detail in the current version, which confuses me is:
* My understanding of the `WALMapper` is:
  * the `Filesystem Chain` provides the blocks
  * the arrow from the top (currently coming from `Badger DB`) provides the state updates 
* But in my head, the data sources are:
  *  `LedgerWAL` provides the state updates 
  *  `Badger DB` provides the blocks 

Could it be that `Badger DB` and `LedgerWAL` should be swapped?

Reply:
> But in my head, the data sources are:
> LedgerWAL provides the state updates
> Badger DB provides the blocks

That is correct, those two "chains" are not related (meaning no contraints are in place). Diagram looks correct to me

Reply:
The arrows are now labelled as to describe what data is being transferred, is it clearer this way?